### PR TITLE
Avoid inserting trailing commas within f-strings

### PR DESCRIPTION
**Pull request by charliermarsh**
Closes https://github.com/astral-sh/ruff/issues/8556.

### CODE REVIEW

File: crates/ruff_linter/resources/test/fixtures/flake8_commas/COM81.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
 4,
 )
 
-foo = 3, 
+foo = 3,
```
Comment (charliermarsh) [inline crates/ruff_linter/resources/test/fixtures/flake8_commas/COM81.py:45]: My editor is stripping this.

**Review by dhruvmanila — APPROVED** [review]
Thanks!

Although this does mean that we wouldn't support similar expressions inside the f-string. This could be a candidate to move the rule to AST checker. We can then use the lexer to check for missing commas for each possible expression.

**Comment by charliermarsh** [comment]
👍 Yeah, I'm not sure we can feasibly support linting within f-strings for this rule as long as it's tokenized.

**Comment by charliermarsh** [comment]
Ahh this is buggy. Need to fix.

**Pull request merged**